src/App.tsx

```diff
@@ -1,24 +1,178 @@
+import { Todo } from './types/Todo';
+import {
+  deleteTodos, getTodos, patchTodo, postTodos,
+} from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { FilterType } from './types/FilterType';
+import { Notification } from './components/Notification';
 
-const USER_ID = 0;
+const USER_ID = 7001;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo>();
+  const [isWaiting, setIsWaiting] = useState(0);
+  const [isDeletingCompleted, setIsDeletingCompleted] = useState(false);
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch {
+        setError('load');
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const addNewTodo = useCallback(async (todo: Omit<Todo, 'id'>) => {
+    setIsDisabledInput(true);
+    setTempTodo({ ...todo, id: 0 });
+    if (todo.title) {
+      try {
+        const newTodo = await postTodos(todo);
+
+        setTodos(state => [...state, newTodo]);
+      } catch {
+        setError('Unable to add a todo');
+        setTimeout(() => {
+          setError('');
+        }, 3000);
+      } finally {
+        setIsDisabledInput(false);
+        setTempTodo(undefined);
+      }
+    } else {
+      setError('Title can`t be empty');
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (id: number) => {
+    setIsWaiting(id);
+
+    try {
+      await deleteTodos(id);
+    } catch {
+      setError('Unable to delete a todo');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      setIsWaiting(0);
+    }
+  }, []);
+
+  const changeTodo = useCallback(async (id: number, data: Partial<Todo>) => {
+    setIsWaiting(id);
+    try {
+      await patchTodo(id, data);
+
+      setTodos(state => state.map(todo => {
+        if (todo.id === id) {
+          return { ...todo, ...data };
+        }
+
+        return todo;
+      }));
+    } catch {
+      setError('Unable to update a todo');
+    } finally {
+      setIsWaiting(0);
+    }
+  }, []);
+
+  const toggleAllTodos = useCallback(() => {
+    const areAllDone = todos.every(todo => todo.completed);
+
+    if (areAllDone) {
+      todos.forEach(el => {
+        changeTodo(el.id, { completed: false });
+      });
+    } else {
+      const notDoneTodos = todos.filter(el => !el.completed);
+
+      notDoneTodos.forEach(element => {
+        changeTodo(element.id, { completed: true });
+      });
+    }
```

Comment:
if you want to disable all loading todos at one time

- pass all ids to array before running async function for each todo
- replace forEach with map
- use Promise.all for this map
- await this promise
- then set empty array to loading todo ids